### ISSUE TITLE
Log file redirection 

### ISSUE THREAD

**Issue by haps-basset** [seud0nym/tch-gui-unhide#157]
This post is not an issue but query/request. Because my modem lately reboots itself or loosing connection, which is why I would like to collect and analyze modem's logs. Unfortunately, after reboot the previous log is whipped out and as such I cannot provide any proof to ISP (Telstra) what is going on. Telstra in its "incredible wisdom" testing lines after an issue saying that he lines are perfect. Well, I am not surprised, but they do not analyze postmortem log of the registered issue. In this circumstance, would it be possible to redirect the log file/system to USB disk in order to keep a track of what is going on?

**Comment by seud0nym** [comment]
If you have root access, the logs are kept in the `/root/log` directory. The `messages.log` file is the current log file, and the previous log file is in `messages.log.0`.

**Comment by haps-basset** [comment]
Many thanks. Yes, I had root access, but I did something silly, and I had to restore to factory settings. Unfortunately, after that, I was not able to get root access again. Anyway, I'll try again.

**Comment by seud0nym** [comment]
Unfortunately, without root access, there isn't much you can do. 

And you would probably find that Telstra wouldn't help you if you get a log from a rooted device. I am pretty sure that they would blame everything on the fact that it was rooted (even though it is nonsense).

**Comment by haps-basset** [comment]
Ya, you are absolutely right, any discussion with Telstra's experts is just pointless. I'm hoping that the last chance to fix the .... problem would be tio.com.au. If I will be lucky to find a tech expert keen to read logs, perhaps then I would be able to force ISP to do for what I am paying. 
I took me little time, but I was able to get root access using your software. As always, many thanks for your support.

**Comment by seud0nym** [comment]
If you want to use the Telstra telephony service, you need to reset the device and take some specific steps:

1. Run `./reset-to-factory-defaults-with-root -d`
2. After it reboots, run `./de-telstra -kT` and any other options you would normally use
3. Then run `./tch-gui-unhide` with whatever options you would normally use.

The `-d` option on `reset-to-factory-defaults-with-root` prevents auto-update of the firmware by Telstra. Normally, we completely prevent Telstra modifying the device by removing the CWMP service that handles that task. But CWMP also configures the device with the telephony settings, so you can't remove or disable it. The `-kT` option on `de-telstra`  says to keep all the Telstra monitoring and logging. If you only want to keep CWMP working but disable all Telstra monitoring, use `-kc` instead. The `-k` options should be specfied _last_ after any other `de-telstra` options you specify.

**Comment by haps-basset** [comment]
Many thanks, I followed your instruction and I got what I wanted, e.g. root access, logs and phone service. 
Once again, many thanks.
Just silly question, I am getting a white light in front of the router, I don't care if it is meaningless message otherwise shall 
I pay attention?

**Comment by seud0nym** [comment]
It probably means that Telstra has attempted to upgrade your firmware because you are not on the latest. It will have failed because you reset using the -d option. You really have 2 choices:

1. Ignore it; or
2. Upgrade to the latest firmware manually.

It is basically the same 3 steps as above, except that you download the latest firmware from the repository to your USB stick and in step 1 add the  `-f firmware.rbi` option, where `firmware.rbi` is the name of the firmware file you download.

**Comment by haps-basset** [comment]
I am happy with white light  👍  Many thanks 🥇

**Comment by haps-basset** [comment]
The reason why I have installed your app was driven by unstable connectivity or modem and needs to find reason of it. After following your suggestion, I got what I wanted, furthermore the modem and connectivity are statable after upgrade!
I am not sure that the improvement is a result of the upgrade, but I am not getting everyday line/modem reboots.
I would like to thank you very much for your support, and the app.
Much appreciated.
